packages/dataviews/package.json

```diff
@@ -48,5 +48,8 @@
 	},
 	"publishConfig": {
 		"access": "public"
+	},
+	"scripts": {
+		"build:wp": "esbuild src/index.ts --bundle --jsx=automatic --outdir=wp --external:react* --external:sprintf-js --external:hoist-non-react-statics --external:use-sync-external-store --external:deepmerge --external:fast-deep-equal --external:highlight-words-core --external:remove-accents --external:equivalent-key-map --external:rungen --external:moment* --external:clsx --external:is-plain-object --external:tslib --external:pascal-case --external:camel-case --external:change-case --external:lower-case --external:no-case --external:dot-case --external:memize --external:@tannin/* --external:tannin --format=esm --tree-shaking=true"
```

Comment:
I spent some time on this and failed to generated the "external" list automatically. There's probably a way though. The idea is that I want to put any package dependency in the nested tree of dependencies as "external" unless it's `@wordpress/*`

Reply:
@jsnajdr @sirreal or anyone else :) Would love help here if anyone has time. I want to instruct esbuild or any other bundler to only bundle `@wordpress/*` dependencies.

Reply:
@luisherranz I know you spent some time exploring esbuild externals, maybe you could help here.

Reply:
[Looking at esbuild's options](https://esbuild.github.io/api/#external), there's nothing about glob negation or a callback form of the option so I'm not sure how this could be done.

[Webpack externals can use a function callback that allows fine control over what's external.](https://webpack.js.org/configuration/externals/#function)

Reply:
What I wanted to do is a custom JS script that reads the package.json and the deep tree of dependencies and extract all dependencies that don't match `@wordpress/*`

Reply:
I'm not sure of the best way to handle that. [DEWP does it to generate the asset files](https://github.com/WordPress/gutenberg/blob/cf59c28e1b0cd0870eb82c21eedf221ebf92c606/packages/dependency-extraction-webpack-plugin/lib/index.js#L173-L180), it relies on webpack. I'm not sure how I'd go about walking the dependency tree, it's not simple.

I guess I'd rely on a bundler to do a first pass to collect all of the imported, then use that list to externalize the desired modules in a second pass.

Reply:
The Esbuild documentation for plugin creation has [a similar example](https://esbuild.github.io/plugins/#on-resolve) of marking imports as externals.

According to that example, it should be possible to do something like this (not tested):

```js
function createWordPressExternalsPlugin() {
  return {
    name: 'wordpress-externals',
    setup(build) {
      build.onResolve({ filter: /^(?!@wordpress\/|\.)/ }, args => {
        return { path: args.path, external: true };
      });
    }
  };
}
```

Reply:
@luisherranz Thanks a lot, I updated the script and now I have something that I believe is working well.